x-pack/plugins/cases/public/components/create/templates.tsx

```diff
@@ -47,7 +56,26 @@ export const TemplateSelectorComponent: React.FC<Props> = ({
       id="createCaseTemplate"
       fullWidth
       label={TEMPLATE_LABEL}
-      labelAppend={OptionalFieldLabel}
+      labelAppend={
+        <EuiFlexGroup
+          alignItems="center"
+          gutterSize="s"
+          css={css`
+            flex-grow: 0;
+          `}
+          responsive={false}
+        >
+          <EuiFlexItem grow={false}>{OptionalFieldLabel}</EuiFlexItem>
+          <EuiFlexItem
+            grow={false}
+            css={css`
+              line-height: 0;
+            `}
+          >
+            <ExperimentalBadge compact={isSmallScreen} />
+          </EuiFlexItem>
```

Comment:
```suggestion
          <EuiFlexItem
            grow={false}
            css={css`
              line-height: 0;
            `}
          >
            <ExperimentalBadge compact={isSmallScreen} />
          </EuiFlexItem>
          <EuiFlexItem grow={false}>{OptionalFieldLabel}</EuiFlexItem>
```

Well, if nothing else works I'd just suggest changing the order here so that the Optionals remain aligned in the sidebar. But I understand it's a nitpick, I'll leave it up to you :D

<img width="1018" alt="Screenshot 2024-07-12 at 10 22 19" src="https://github.com/user-attachments/assets/aac4140e-8ca2-4869-8f86-c9f255013e75">

Reply:
I also think that it would be nice to show `Technical Preview` badge before optional.

Reply:
Makes perfect sense @adcoelho 🙂